ui/apps/platform/src/Containers/Collections/parser.ts

```diff
@@ -67,14 +80,36 @@ export function parseCollection(data: CollectionResponse): Collection | Aggregat
             return;
         }
 
-        if (!collection.selectorRules[entity]) {
-            collection.selectorRules[entity] = {
-                field,
-                rules: [],
-            };
+        if (isEmpty(collection.selectorRules[entity])) {
+            if (isByLabelField(field)) {
+                collection.selectorRules[entity] = {
+                    field,
+                    rules: [],
+                };
+            } else if (isByNameField(field)) {
+                collection.selectorRules[entity] = {
+                    field,
+                    rule: { operator: 'OR', values: [] },
+                };
+            }
         }
 
-        collection.selectorRules[entity]?.rules.push(rule);
+        const selector = collection.selectorRules[entity];
+
+        if (isByLabelSelector(selector)) {
+            const firstValue = rule.values[0]?.value;
+
+            if (firstValue && firstValue.includes(LABEL_SEPARATOR)) {
+                const key = firstValue.split(LABEL_SEPARATOR)[0] ?? '';
+                selector.rules.push({
+                    operator: 'OR',
+                    key,
+                    values: rule.values.map(({ value }) => value.split('=')[1] ?? ''),
```

Comment:
Remind me, is there, will there, or even can there be form validation to limit valid characters.

Just today, we saw an intentional fuzzing attempt to include split character in a value.

Reply:
Initially I was using the label helper methods on user input to ensure these were valid k8s labels, but then realized that these form fields need to support RE2 syntax.

Either way, I'll make a note to follow up with the BE team and implement something more robust here.